modular_skyrat/modules/synths/code/surgery/robot_healing.dm

```diff
@@ -0,0 +1,162 @@
+//Almost copypaste of tend wounds, with some changes
+/datum/surgery/robot_healing
+	steps = list(
+		/datum/surgery_step/mechanic_open,
+		/datum/surgery_step/pry_off_plating,
+		/datum/surgery_step/cut_wires,
+		/datum/surgery_step/robot_heal,
+		/datum/surgery_step/mechanic_close,
+	)
+
+	target_mobtypes = list(/mob/living/carbon/human)
+	possible_locs = list(BODY_ZONE_CHEST)
+	replaced_by = /datum/surgery
+	requires_bodypart_type = BODYTYPE_ROBOTIC
+	surgery_flags = SURGERY_IGNORE_CLOTHES | SURGERY_REQUIRE_RESTING | SURGERY_REQUIRE_LIMB
+
+	/// The step to use in the 4th surgery step.
+	var/healing_step_type
+	/// If true, doesn't send the surgery preop message. Set during surgery.
+	var/surgery_preop_message_sent = FALSE
+
+/datum/surgery/robot_healing/New(surgery_target, surgery_location, surgery_bodypart)
+	..()
+	if(healing_step_type)
+		steps = list(
+			/datum/surgery_step/mechanic_open,
+			/datum/surgery_step/pry_off_plating,
+			/datum/surgery_step/cut_wires,
+			healing_step_type,
+			/datum/surgery_step/mechanic_close,
+		)
+
+/datum/surgery_step/robot_heal
+	name = "repair body (welder/cable)"
+	implements = list(TOOL_WELDER = 100, /obj/item/stack/cable_coil = 100)
+	repeatable = TRUE
+	time = 25
+
+	/// If this surgery is healing brute damage. Set during operation steps.
+	var/heals_brute = FALSE
+	/// If this surgery is healing burn damage. Set during operation steps.
+	var/heals_burn = FALSE
+	/// How much healing the sugery gives.
+	var/brute_heal_amount = 0
+	/// How much healing the sugery gives.
+	var/burn_heal_amount = 0
+	/// Heals an extra point of damage per X missing damage of type (burn damage for burn healing, brute for brute). Smaller number = more healing!
+	var/missing_health_bonus = 0
+
+/datum/surgery_step/robot_heal/tool_check(mob/user, obj/item/tool)
+	if(implement_type == TOOL_WELDER && !tool.tool_use_check(user, 1))
+		return FALSE
+	return TRUE
+
+/datum/surgery_step/robot_heal/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/woundtype
+	if(implement_type == TOOL_WELDER)
+		heals_brute = TRUE
+		heals_burn = FALSE
+		woundtype = "dents"
+	else
+		heals_brute = FALSE
+		heals_burn = TRUE
+		woundtype = "wiring"
+
+	if(istype(surgery,/datum/surgery/robot_healing))
+		var/datum/surgery/robot_healing/the_surgery = surgery
+		if(!the_surgery.surgery_preop_message_sent)
+			display_results(
+				user,
+				target,
+				span_notice("You attempt to fix some of [target]'s [woundtype]."),
+				span_notice("[user] attempts to fix some of [target]'s [woundtype]."),
+				span_notice("[user] attempts to fix some of [target]'s [woundtype]."),
+			)
+
+/datum/surgery_step/robot_heal/initiate(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, try_to_fail = FALSE)
+	if(..())
+		while((heals_brute && target.getBruteLoss() && tool.tool_use_check(user,1)) || (heals_burn && target.getFireLoss() && tool))
+			if(!..())
+				break
```

Comment:
I kind of hate this.
That just feels wrong, but I'm not sure if there's really a better way to do it, eugh.

Reply:
I... yea. I took one look and went "I'm not touching that yet."